plugins/analysis-phonetic/src/main/java/org/elasticsearch/index/analysis/phonetic/FrenchPhonetic.java

```diff
@@ -0,0 +1,287 @@
+package org.elasticsearch.index.analysis.phonetic;
+
+import org.apache.commons.codec.EncoderException;
+import org.apache.commons.codec.StringEncoder;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * <p>French phonetic plugin has been developped to increase number of matching words based on the way they are pronounced in French.</p>
+ * </p>All current latin phonetic plugins matched too much french words as equal.</p>
+ * </p>Not all phonemes have been coded because it would have been too restrictive.</p>
+ *
+ * <pre>
+ *     1 for in sound
+ *     2 for &eacute; sound
+ *     3 for an sound
+ *     4 for on sound
+ *     8 for oeu/eu sound
+ * </pre>
+ *
+ * @see <a href="https://fr.wikipedia.org/wiki/Prononciation_du_fran%C3%A7ais#Prononciation_des_graph.C3.A8mes"/> French prononciation on Wikipedia</a>
+ *
+ * @author Harold Capitaine
+ * @author Galeries Lafayette
+ */
+public class FrenchPhonetic implements StringEncoder {
+
+    private static final List<Character> VOWELS = Arrays.asList('A', 'E', '2', 'I', 'O', 'U', 'Y');
```

Comment:
2 is a french sound made only of vowels and for example baisé and baissé
would have been equally translated as B2S2 but not as B2Z2 and B2S2 if 2
was not included in the constant.

Worked mainly with all case VOWELS constant is used in.

If you find a non working case I will put the 2 in if statements where
needed and remove it from the constant.

Contribute to the tests ;)

On Thu, Jan 14, 2016 at 8:34 PM, David Pilato <notifications@github.com>
wrote:

> In
> plugins/analysis-phonetic/src/main/java/org/elasticsearch/index/analysis/phonetic/FrenchPhonetic.java
> <https://github.com/elastic/elasticsearch/pull/15996#discussion_r49773295>
> :
>
> > + * <pre>
> > + *     1 for in sound
> > + *     2 for &eacute; sound
> > + *     3 for an sound
> > + *     4 for on sound
> > + *     8 for oeu/eu sound
> > + * </pre>
> > + *
> > + * @see <a href="https://fr.wikipedia.org/wiki/Prononciation_du_fran%C3%A7ais#Prononciation_des_graph.C3.A8mes"/> French prononciation on Wikipedia</a>
> > + *
> > + * @author Harold Capitaine
> > + * @author Galeries Lafayette
> > + */
> > +public class FrenchPhonetic implements StringEncoder {
> > +
> > +    private static final List<Character> VOWELS = Arrays.asList('A', 'E', '2', 'I', 'O', 'U', 'Y');
>
> Why 2 here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/15996/files#r49773295>.
>